Planner/app/src/main/java/rdatu/android/cyscorpions/com/projectplanner/view/ScheduleTaskActivity.java

```diff
@@ -0,0 +1,278 @@
+                        finish();
+                        return;
+                    }
+
+                    for (int i = timeStart; i < timeEnd; i++) {
+                        try {
+                            hasDuplicate = mTaskManager.hasTasks(date, getFormattedTime(i));
+                            if (hasDuplicate)
+                                break;
+                        } catch (Exception e) {
+                            Log.e("Planner", "Something went wrong!", e);
+                        }
+                    }
+
+                    if (hasDuplicate) {
+                        showOverwriteDialog(name, descr, place, priority, date, timeStart, timeEnd);
+                    } else {
+                        for (int i = timeStart; i < timeEnd; i++) {
+                            saveTasks(date, name, descr, place, priority, getFormattedTime(i));
+                        }
+                    }
+
+                } else {
+                    Toast.makeText(getApplicationContext(), "All fields are required.", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
+
+
+    private void saveTasks(String date, String name, String descr, String place, String priority, String time) {
+        Tasks task = new Tasks();
+        task.setDate(date);
+        task.setTaskName(name);
+        task.setDescription(descr);
+        task.setPlace(place);
+        task.setPriority(priority);
+        task.setTimeSlot(time);
+        mTaskManager.saveTask(task);
+        finish();
+
+    }
+
+    private void showDatePicker() {
+        FragmentManager fm = getSupportFragmentManager();
+        DatePickerDialog dialog = DatePickerDialog.newInstance(mDateButton.getText().toString(), FUNCTION_FORCHANGE);
+        dialog.show(fm, "datePicker");
+    }
+
+    private void showTimePicker() {
+        FragmentManager fm = getSupportFragmentManager();
+        TimePickerDialog dialog = TimePickerDialog.newInstance(mToTimeText.getText().toString());
+        dialog.show(fm, "timePicker");
+    }
+
+    private void showOverwriteDialog(String name, String descr, String place, String priority, String date, int start, int end) {
+        FragmentManager fm = getSupportFragmentManager();
+        OverwriteDialog dialog = OverwriteDialog.newInstance(name, descr, place, priority, date, start, end);
+        dialog.show(fm, "overWrite");
+    }
+
+    @Override
+    public void onTimeChanged(String time) {
+        mToTimeText.setText(time);
+        mTimeEnd = mToTimeText.getText().toString();
+    }
+
+    @Override
+    public void onDateChanged(String date) {
+        mDateButton.setText(date);
+    }
+
+    @Override
+    public void onJumpTo(String date) {
+        //Not used in this Activity
+    }
+
+    private boolean areFieldsFilled() {
+        boolean ready = true;
+
+        ready = isNotEmpty(mTaskNameText);
+        ready = isNotEmpty(mTaskDescriptionText);
+        ready = isNotEmpty(mPlaceText);
```

Comment:
This line of code might be wrong? For example if ```mTaskNameText``` is empty (ready = false) already and then ```mTaskDescriptionText``` is empty then ready will become true even if its supposed to be false because of the first condition.

We should refactor the code and also include the conditions below.